LanguageFeatures/Inline-classes/static_analysis_inline_class_A05_t03.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Assume that T1, .. Ts are types, and V resolves to an inline
+/// class declaration of the following form:
+///
+/// inline class V<X1 extends B1, .. Xs extends Bs> ... {
+///   final T id;
+///   V(this.id);
+///
+///   ... // Other members.
+/// }
+/// ...
+/// When s is zero, V<T1, .. Ts> simply stands for V, a non-generic inline type.
+/// When s is greater than zero, a raw occurrence V is treated like a raw type:
+/// Instantiation to bound is used to obtain the omitted type arguments
+///
+/// @description Checks that instantiation to bound is used to obtain the
+/// omitted type arguments
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/static_type_helper.dart";
+
+inline class V1<T extends num> {
+  final int id;
+  V1(this.id);
+}
+
+inline class V2<T extends Object> {
+  final int id;
+  V2(this.id);
+}
+
+inline class V3<T extends Object?> {
+  final int id;
+  V3(this.id);
+}
```

Comment:
Looks fine!

Yes, please, it would be nice to have a test that confirms that some of those type arguments are reported as errors (unless that is just a duplicate of existing tests about being regular-bounded).

You can create a type that satisfies the bound: `class B extends A<B> {}`, and then use `V<B>`. That's the standard way to obtain a type that satisfies an F-bound: Just write the F-bound itself as a class declaration (`T extends A<T>` becomes `class B extends A<B> {}`).

Reply:
Thanks!